File: pyESN/mackey.py
import numpy as np
import pandas as pd
from pyESN import ESN
from matplotlib import pyplot as plt

data = np.load('mackey_glass_t17.npy') #  http://minds.jacobs-university.de/mantas/code
esn = ESN(n_inputs = 1,
          n_outputs = 1,
          n_reservoir = 500,
          spectral_radius = 1.5,
          random_state=42)
          
trainlen = 2000
future = 2000
pred_training = esn.fit(np.ones(trainlen),data[:trainlen])

prediction = esn.predict(np.ones(future))
print("test error: \n"+str(np.sqrt(np.mean((prediction.flatten() - data[trainlen:trainlen+future])**2))))

plt.figure(figsize=(11,1.5))
plt.plot(range(0,trainlen+future),data[0:trainlen+future],'k',label="target system")
plt.plot(range(trainlen,trainlen+future),prediction,'r', label="free running ESN")
lo,hi = plt.ylim()
plt.plot([trainlen,trainlen],[lo+np.spacing(1),hi-np.spacing(1)],'k:')
plt.legend(loc=(0.61,1.1),fontsize='x-small')
plt.show()

File: csESN/Mackey.cs
using System;
using NumpyDotNet;

namespace csESN
{
    public class Mackey
    {
        public static void Run()
        {
            var data = np.fromfile("./data/mackey_glass_t17.npy", dtype: np.Float64); //  http://minds.jacobs-university.de/mantas/code
            data = (ndarray)data["10:"]; // hack or bug. mackey_glass_t17.npy was saved by numpy. 
                                         // NumpyDotNet np.fromfile load first 10 itens that are not loaded by numpy np.load.
            var esn = new ESN(n_inputs: 1,
                      n_outputs: 1,
                      n_reservoir: 500,
                      spectral_radius: 1.5,
                      random_state: 42);

            int trainlen = 2000;
            int future = 2000;
            var pred_training = esn.fit(np.ones(trainlen), (ndarray)data[$":{trainlen}"]);
            var prediction = esn.predict(np.ones(future));
            ndarray sqr = np.power(prediction.flatten() - data[$"{trainlen}:{trainlen + future}"], 2);
            System.Console.WriteLine("test error: \n" + np.sqrt((ndarray)np.mean(sqr)));
        }
    }
}